File: Doubao_MarsCode/java/MC0003/Main.java
package MC0003;

/**
 * 3. 数字字符串格式化
 * 模拟
 */
public class Main {
    public static String solution(String s) {
        // 去除前导零
        s = s.replaceFirst("^0+(?!$)", "");

        // 分离整数和小数部分
        String[] parts = s.split("\\.");
        String integerPart = parts[0];
        String decimalPart = parts.length > 1 ? "." + parts[1] : "";

        // 格式化整数部分
        StringBuilder formattedInteger = new StringBuilder();
        int count = 0;
        for (int i = integerPart.length() - 1; i >= 0; i--) {
            formattedInteger.append(integerPart.charAt(i));
            count++;
            if (count % 3 == 0 && i != 0) {
                formattedInteger.append(",");
            }
        }
        formattedInteger.reverse();

        // 合并整数和小数部分
        return formattedInteger.toString() + decimalPart;
    }

    public static void main(String[] args) {
        // Add your test cases here
        System.out.println(solution("1294512.12412").equals("1,294,512.12412"));
        System.out.println(solution("0000123456789.99").equals("123,456,789.99"));
        System.out.println(solution("987654321").equals("987,654,321"));
    }
}

File: Doubao_MarsCode/go/MC0003/main.go
package main

import (
	"fmt"
	"strings"
)

func solution(s string) string {
	// 去除前导零
	s = strings.TrimLeft(s, "0")
	if s == "" {
		s = "0"
	}

	// 分离整数和小数部分
	parts := strings.Split(s, ".")
	integerPart := parts[0]
	decimalPart := ""
	if len(parts) > 1 {
		decimalPart = "." + parts[1]
	}

	// 格式化整数部分
	var formattedInteger strings.Builder
	count := 0
	for i := len(integerPart) - 1; i >= 0; i-- {
		formattedInteger.WriteByte(integerPart[i])
		count++
		if count%3 == 0 && i != 0 {
			formattedInteger.WriteByte(',')
		}
	}

	// 反转格式化的整数部分
	runes := []rune(formattedInteger.String())
	for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
		runes[i], runes[j] = runes[j], runes[i]
	}

	return string(runes) + decimalPart
}

func main() {
	fmt.Println(solution("1294512.12412") == "1,294,512.12412")
	fmt.Println(solution("0000123456789.99") == "123,456,789.99")
	fmt.Println(solution("987654321") == "987,654,321")
}
